File: CS/DataModel.cs
using System;
using System.Collections.ObjectModel;

namespace ChildNodesSelector {
    public class BaseObject {
        public string Name { get; set; }
        public string Executor { get; set; }
        public string State { get; set; }
    }

    public class ProjectObject : BaseObject {
        public ObservableCollection<ProjectStage> Stages { get; set; }
    }

    public class ProjectStage : BaseObject {
        public ObservableCollection<StageTask> Tasks { get; set; }
    }

    public class StageTask : BaseObject {
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
    }
}

File: VB/DataModel.vb
Imports System.Collections.ObjectModel

Namespace ChildNodesSelector

    Public Class BaseObject

        Public Property Name As String

        Public Property Executor As String

        Public Property State As String
    End Class

    Public Class ProjectObject
        Inherits BaseObject

        Public Property Stages As ObservableCollection(Of ProjectStage)
    End Class

    Public Class ProjectStage
        Inherits BaseObject

        Public Property Tasks As ObservableCollection(Of StageTask)
    End Class

    Public Class StageTask
        Inherits BaseObject

        Public Property StartDate As Date

        Public Property EndDate As Date
    End Class
End Namespace
